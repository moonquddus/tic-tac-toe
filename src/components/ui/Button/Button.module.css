.button {
  border: none;
  outline: none;
  box-shadow: none;
  cursor: pointer;
  color: var(--button-color, var(--color-black));
  background: var(--button-background, var(--color-white));
  border: solid 1px var(--button-border, var(--color-grey-1));
  border-radius: var(--border-radius);
  padding: 5px 15px;
  font-size: 1.2rem;
  line-height: 1.6rem;
}

.button:hover {
  background: var(--button-background-hover, var(--button-background));
}

.button:focus,
.button:focus-within,
.button-active {
  outline: solid 3px var(--color-black);
}

.button[data-variation=primary] {
  --button-background: var(--color-green-1);
  --button-background-hover: var(--color-green-2);
  --button-color: var(--color-white);
  --button-border-color: var(--color-grey-2);
}