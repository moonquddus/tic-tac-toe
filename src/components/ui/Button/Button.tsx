import { ButtonHTMLAttributes, PropsWithChildren } from 'react'
import styles from './Button.module.css'

/**
 * There is plenty more I can do here
 * eg. more themes, sizes, a disabled state, etc
 */

interface ButtonProps extends PropsWithChildren<ButtonHTMLAttributes<HTMLButtonElement>> {
  variation?: 'primary' | 'secondary'
}

function Button(props: ButtonProps) {
  const {children, className, variation = 'primary', ...data} = props
  return (
    <button {...data} className={`${styles.button} ${className}`} data-variation={variation}>
      {children}
    </button>
  )
}

export default Button
