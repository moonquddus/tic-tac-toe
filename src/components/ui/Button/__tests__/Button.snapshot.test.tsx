import { render } from '@testing-library/react'
import Button from '../Button'

describe('Button label', () => {
  it('should render children as Button text', () => {
    const wrapper = render(<Button>Hello I am a button</Button>)
    expect(wrapper.baseElement).toMatchSnapshot()
  })
})

describe('Variations', () => {
  it('should render a primary button', () => {
    const wrapper = render(<Button variation='primary'>Primary button</Button>)
    expect(wrapper.baseElement).toMatchSnapshot()
  })

  it('should render a secondary button', () => {
    const wrapper = render(<Button variation='secondary'>Secondary button</Button>)
    expect(wrapper.baseElement).toMatchSnapshot()
  })
})
