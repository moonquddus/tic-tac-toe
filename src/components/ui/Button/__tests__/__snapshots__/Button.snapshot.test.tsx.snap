// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button label should render children as Button text 1`] = `
<body>
  <div>
    <button
      class="button undefined"
      data-variation="primary"
    >
      Hello I am a button
    </button>
  </div>
</body>
`;

exports[`Variations should render a primary button 1`] = `
<body>
  <div>
    <button
      class="button undefined"
      data-variation="primary"
    >
      Primary button
    </button>
  </div>
</body>
`;

exports[`Variations should render a secondary button 1`] = `
<body>
  <div>
    <button
      class="button undefined"
      data-variation="secondary"
    >
      Secondary button
    </button>
  </div>
</body>
`;
